examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -604,7 +604,7 @@ static void wfx_rsi_save_ap_info(void) // translation
     sl_wifi_ssid_t ssid_arg;
     memset(&ssid_arg, 0, sizeof(ssid_arg));
     ssid_arg.length = wfx_rsi.sec.ssid_length;
```

Comment:
for `sl_wifi_ssid_t` ` .value` buffer is of size 32 while `wfx_rsi.sec `is a `wfx_wifi_provision_t` struct where `wfx_rsi.sec.ssid` is of size 33 (WFX_MAX_SSID_LENGTH+1) . we have a risk of out of bound write with the following copystring.

can sl_wifi_ssid_t .value be increased to (WFX_MAX_SSID_LENGTH+1). Can we add a check that the resulting ssid_arg.length is at max size of`  ssid_arg.value - 1`

Reply:
@jmartinez-silabs sl_wifi_ssid_t struct value is in wifi sdk and it cannot be modified. 
I was kinda thinking to add a check for time being, something like
`ssid_arg.length = chip::min<size_t>(wfx_rsi.sec.ssid_length, WFX_MAX_SSID_LENGTH - 1);`
This will ensure the copy string will of ssid_arg.value never overflow i.e., get 32 max
`chip::Platform::CopyString((char *) &ssid_arg.value[0], ssid_arg.length + 1, wfx_rsi.sec.ssid); // +1 for null termination`
Once it gets fixed in the wifi sdk then this check can be removed

Reply:
since `ssid_arg.value` is not linked in any way to `WFX_MAX_SSID_LENGTH` and it is unrelatedly set to 32, this is situationally true. If `WFX_MAX_SSID_LENGTH` value changes this becomes invalid and it will silently break at an unknown time.

the correct approach would be 
`ssid_arg.length = chip::min<size_t>(wfx_rsi.sec.ssid_length, sizeof(ssid_arg.value) - 1)`

Reply:
Makes sense, updated the PR. Thanks